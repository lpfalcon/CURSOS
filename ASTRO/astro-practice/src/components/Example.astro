---
import Layout from "../layouts/Layout.astro";
---

<Layout title={Astro.props.title}>
  <section class="my-24">
    <div class="px-10">
      <slot />
    </div>
  </section>
</Layout>
