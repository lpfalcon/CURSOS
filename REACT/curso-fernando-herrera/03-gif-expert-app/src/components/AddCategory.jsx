import { useState } from "react";
import PropTypes from 'prop-types';

export const AddCategory = ( { onNewCategory } ) => {

    const [inputValue, setInputValue] = useState('')

    const onInputChanged = ({ target }) => {

        setInputValue(target.value )

    }

    const onSubmit = ( event ) => {
        event.preventDefault()
        const newInputValue = inputValue.trim()
        if( !newInputValue.length ) return
        
        onNewCategory( newInputValue )
        setInputValue('')

    }

    return (
        <form onSubmit={ onSubmit } aria-label="form">

            <input 
                type="text"
                placeholder="Buscar Gif"
                value={ inputValue }
                onChange={ onInputChanged }
            />

        </form>
    )
}

AddCategory.propTypes = {

    onNewCategory : PropTypes.func.isRequired

}
