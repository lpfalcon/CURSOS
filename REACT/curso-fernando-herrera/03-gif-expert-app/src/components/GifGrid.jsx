
import { useFetchGifs } from "../hooks/useFetchGifs";
import { GifItem } from "./GifItem";
import PropTypes from "prop-types";

 export const GifGrid = ({ category }) => {
 
    const { images, isLoading } = useFetchGifs(category)

    return (
        <div>
            <h3> {category} </h3>
            {
                isLoading && ( <h2 >  Cargando ... </h2>)
            }
           
            <div className="card-grid">
                {
                    images.map(( image ) => (

                       <GifItem
                         key={ image.id } 
                         title={ image.title }
                         {... image }
                        />
                    ))
                }

            </div>
        </div>
    );

}
GifGrid.propTypes = {

    category : PropTypes.string.isRequired

}
