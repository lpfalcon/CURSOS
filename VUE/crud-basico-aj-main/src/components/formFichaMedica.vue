<template>
  <MiddleBox class="shadowHover centerDiv">
    <v-form v-model="formValid" @submit.prevent>
      <v-container>
        <v-row>
          <v-col cols="12" md="12">
            <h1>{{ titulo }}</h1>
          </v-col>

          <v-col cols="12" md="12">
            <slot />
          </v-col>

          <v-col cols="12" md="4">
            <v-text-field
              :rules="rules"
              label="Nombre"
              v-model="formData.name"
              required
            ></v-text-field>
          </v-col>

          <v-col cols="12" md="4">
            <v-text-field
              :rules="rules"
              label="Precio"
              v-model="formData.price"
              type="number"
              required
            ></v-text-field>
          </v-col>

          <v-col cols="12" md="4">
            <v-text-field
              :rules="rules"
              label="Id"
              v-model="formData.id"
              required
            ></v-text-field>
          </v-col>




          <v-col cols="12" md="12">
            <v-btn
              color="success"
              class="mr-4"
              type="submit"
              :disabled="!formValid"
              @click="enviarDataChildren()"
            >
              Registrar
            </v-btn>

            <v-btn color="error" class="mr-4" @click="reset()" v-if="limpiar">
              Limpiar
            </v-btn>
          </v-col>
        </v-row>
      </v-container>
    </v-form>
  </MiddleBox>
</template>

<script>
import MiddleBox from "../components/static/MiddleBox.vue";

export default {
  name: "HomeView",
  components: {
    MiddleBox,
  },
  props: {
    formData: Object,
    titulo: String,
    limpiar: Boolean,
  },
  data: () => ({
    formDataDefault: {
      name: "",
      price: "",
      id: "",
    },
    formValid: false,
    rules: [
      (value) => {
        if (value) return true;

        return "Este campo es obligatorio";
      },
    ],
  }),
  methods: {
    enviarDataChildren: function () {
      this.$emit("enviarData");
    },
    reset: function () {
      this.formData = this.formDataDefault
    },
  },
};
</script>

<style scoped>
.home {
  background: #fff;
}

.centerDiv {
  position: absolute;
  top: 50%;
  left: 50%;
  transform: translate(-50%, -50%);
}
</style>
