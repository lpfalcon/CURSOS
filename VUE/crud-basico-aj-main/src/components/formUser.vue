<template>
  <MiddleBox class="shadowHover centerDiv">
    <v-form v-model="formValid" @submit.prevent>
      <v-container>
        <v-row>
          <v-col cols="12" md="12">
            <h1>{{ titulo }}</h1>
          </v-col>

          <v-col cols="12" md="12">
            <slot />
          </v-col>

          <v-col cols="12" md="4">
            <v-text-field
              :rules="rules"
              label="Nombre"
              v-model="formData.nombre"
              required
            ></v-text-field>
          </v-col>

          <v-col cols="12" md="4">
            <v-text-field
              :rules="rules"
              label="Correo"
              type="email"
              v-model="formData.correo"
              required
            ></v-text-field>
          </v-col>

          <v-col cols="12" md="4">
            <v-text-field
              :rules="rules"
              type="password"
              label="ContraseÃ±a"
              v-model="formData.password"
              required
            ></v-text-field>
          </v-col>


          <v-col cols="12" md="12">
            <v-btn
              color="success"
              class="mr-4"
              type="submit"
              :disabled="!formValid"
              @click="enviarDataChildren()"
            >
              Registrar
            </v-btn>

            <v-btn color="error" class="mr-4" @click="reset()" v-if="limpiar">
              Limpiar
            </v-btn>
          </v-col>
        </v-row>
      </v-container>
    </v-form>
  </MiddleBox>
</template>

<script>
import MiddleBox from "../components/static/MiddleBox.vue";

export default {
  name: "HomeView",
  components: {
    MiddleBox,
  },
  props: {
    formData: Object,
    titulo: String,
    limpiar: Boolean,
  },
  data: () => ({
    formDataDefault: {
      nombre: "",
      correo: "",
      password: ""
    },
    formValid: false,
    rules: [
      (value) => {
        if (value) return true;

        return "Este campo es obligatorio";
      },
    ],
  }),
  methods: {
    enviarDataChildren: function () {
      this.$emit("enviarData");
    },
    reset: function () {
      this.formData = this.formDataDefault
    },
  },
};
</script>

<style scoped>
.home {
  background: #fff;
}

.centerDiv {
  margin-top: 50px !important;
}
</style>
