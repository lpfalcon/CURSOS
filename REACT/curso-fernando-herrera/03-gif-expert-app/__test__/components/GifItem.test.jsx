import {  render, screen } from '@testing-library/react';
import { GifItem } from '../../src/components/GifItem';

describe('<GifItem/> Component Testing', () => {
    const title = 'title';
    const url = 'http://localhost/url';

    test('Should match with the snapshot', () => {
       
        const { container } =  render(<GifItem title={title} url={url} />);
        expect(container).toMatchSnapshot();

    })

    test('Should show the image and the alt text', () => {

        render(<GifItem title={title} url={url} />);
        // screen.debug()
        // console.log(screen.getByRole('img').title)
        // expect(screen.getByRole('img').src).toBe(url)
        // expect(screen.getByRole('img').alt).toBe(title)

        const { src, alt } = screen.getByRole('img');
        expect(src).toBe(url);
        expect(alt).toBe(title);

    })

    test('Shold be show the title in the component', ()=>{
        
        render(<GifItem title={title} url={url} />);
        expect( screen.getByText( title ) ).toBeTruthy();

    })
})