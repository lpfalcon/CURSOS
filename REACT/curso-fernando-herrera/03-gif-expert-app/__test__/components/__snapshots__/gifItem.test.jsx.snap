// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GifItem/> Component Testing Should match with the snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="title"
      src="http://localhost/url"
    />
    <p>
      title
    </p>
  </div>
</div>
`;
